crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -225,22 +235,165 @@ where
+                                        .await
+                                        .entry(block_hash)
+                                        .or_default()
+                                        .extend(all_logs);
+                                }
+                            }
+                        }
+                    }
+                    FilterBlockOption::Range { ref from_block, ref to_block } => {
+                        if let (Some(from), Some(to)) = (
+                            from_block.and_then(|from| from.as_number()),
+                            to_block.and_then(|to| to.as_number()),
+                        ) {
+                            if from > to {
+                                continue
+                            }
+
+                            for block_number in from..=to {
+                                if let Some(block) = old_blocks.get(&block_number) {
+                                    let block_num_hash =
+                                        BlockNumHash::new(block_number, block.header.hash());
+                                    if let Ok(Some(receipts)) =
+                                        self.inner.receipts(block_num_hash).await
+                                    {
+                                        if let Some(receipts) = Arc::into_inner(receipts) {
+                                            let all_logs = get_removed_block_logs(
+                                                block.clone(),
+                                                &FilteredParams::new(Some(*filter.clone())),
+                                                block_num_hash,
+                                                &receipts,
+                                                &self.inner.provider,
+                                            );
+                                            if all_logs.is_empty() {
+                                                continue;
+                                            }
+
+                                            active_filter
+                                                .reorg_blocks
+                                                .write()
+                                                .await
+                                                .entry(block_num_hash.hash)
+                                                .or_default()
+                                                .extend(all_logs);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
     }
 }
 
+/// Get removed logs of a block's receipts.
+fn get_removed_block_logs<P: BlockReader>(
+    block: SealedBlockWithSenders,
+    filter: &FilteredParams,
+    block_num_hash: BlockNumHash,
+    receipts: &[Receipt],
+    provider: &P,
```

Comment:
this shouldn't be required I believe because we already have everything for that block